custom_components/localtuya/common.py

```diff
@@ -55,29 +59,38 @@ async def async_setup_entry(
     This is a generic method and each platform should lock domain and
     entity_class with functools.partial.
     """
-    tuyainterface, entities_to_setup = prepare_setup_entities(
-        hass, config_entry, domain
-    )
-    if not entities_to_setup:
-        return
-
-    dps_config_fields = list(get_dps_for_platform(flow_schema))
-
     entities = []
-    for device_config in entities_to_setup:
-        # Add DPS used by this platform to the request list
-        for dp_conf in dps_config_fields:
-            if dp_conf in device_config:
-                tuyainterface.dps_to_request[device_config[dp_conf]] = None
-
-        entities.append(
-            entity_class(
-                tuyainterface,
-                config_entry,
-                device_config[CONF_ID],
-            )
-        )
 
+    for dev_id in config_entry.data[CONF_DEVICES]:
+        # entities_to_setup = prepare_setup_entities(
+        #     hass, config_entry.data[dev_id], domain
+        # )
+        dev_entry = config_entry.data[CONF_DEVICES][dev_id]
+        entities_to_setup = [
+            entity
+            for entity in dev_entry[CONF_ENTITIES]
+            if entity[CONF_PLATFORM] == domain
+        ]
+
+        if len(entities_to_setup) > 0:
```

Comment:
why this? maybe directly` if entities_to_setup`

Reply:
Well, this is because I'm not a native python programmer (I am more a C++ coder), so I don't know all these tricks... thank you for the suggestion! 😉